diff --git a/plasma/shells/io.gitgud.wackyideas.desktop/contents/lockscreen/AuthUI.qml b/plasma/shells/io.gitgud.wackyideas.desktop/contents/lockscreen/AuthUI.qml
index 57dcb038..d45c954b 100644
--- a/plasma/shells/io.gitgud.wackyideas.desktop/contents/lockscreen/AuthUI.qml
+++ b/plasma/shells/io.gitgud.wackyideas.desktop/contents/lockscreen/AuthUI.qml
@@ -96,7 +96,10 @@ Item {
     Connections {
         target: authenticator
 
-        function onFailed() {
+        function onFailed(kind) {
+            if (kind != 0) { // if this is coming from the noninteractive authenticators
+                return;
+            }
             showMessage(i18nd("kscreenlocker_greet", "The user name or password is incorrect."), "dialog-error");
         }
         function onBusyChanged() {
@@ -112,7 +115,12 @@ Item {
             showMessage(authenticator.errorMessage, "dialog-error");
         }
         function onPromptForSecretChanged() {
-            authenticator.respond(password.text);
+            if (!authenticator.promptForSecret) return;
+            Qt.callLater(function() {
+                if (!authenticator.busy) {
+                    authenticator.respond(password.text);
+                }
+            });
         }
         function onSucceeded() {
             lockSuccess.play();
@@ -282,7 +290,10 @@ Item {
                             enabled: loginMa.enabled
                             font.pointSize: 9
                             text: PasswordSync.password
-                            padding: 4
+                            leftPadding: 8
+                            rightPadding: 8
+                            topPadding: 4
+                            bottomPadding: 4
                             placeholderText: i18nd("plasma_lookandfeel_org.kde.lookandfeel", "Password")
                             echoMode: TextInput.Password
                             inputMethodHints: Qt.ImhHiddenText | Qt.ImhSensitiveData | Qt.ImhNoAutoUppercase | Qt.ImhNoPredictiveText
